qiskit/aqua/operators/state_fns/state_fn.py

```diff
@@ -0,0 +1,353 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+""" StateFn Class """
+
+from typing import Union, Optional, Callable, Set, Dict
+import numpy as np
+
+from qiskit.quantum_info import Statevector
+from qiskit.result import Result
+from qiskit import QuantumCircuit
+from qiskit.circuit import Instruction, ParameterExpression
+
+from ..operator_base import OperatorBase
+
+
+class StateFn(OperatorBase):
+    r"""
+    A class for representing state functions and measurements.
+
+    State functions are defined to be complex functions over a single binary string (as
+    compared to an operator, which is defined as a function over two binary strings, or a
+    function taking a binary function to another binary function). This function may be
+    called by the eval() method.
+
+    Measurements are defined to be functionals over StateFns, taking them to real values.
+    Generally, this real value is interpreted to represent the probability of some classical
+    state (binary string) being observed from a probabilistic or quantum system represented
+    by a StateFn. This leads to the equivalent definition, which is that a measurement m is
+    a function over binary strings producing StateFns, such that the probability of measuring
+    a given binary string b from a system with StateFn f is equal to the inner
+    product between f and m(b).
+
+    NOTE: State functions here are not restricted to wave functions, as there is
+    no requirement of normalization.
+    """
+
+    @staticmethod
+    # pylint: disable=unused-argument,inconsistent-return-statements
```

Comment:
inconsistent return can again be removed by raising an exception if none of the isinstances match rather than just falling off the end of the function

Reply:
Fixed